common/include/villas/kernel/devices/device_connection.hpp

```diff
@@ -0,0 +1,79 @@
+/* Vfio connection to a device.
+ *
+ * Author: Pascal Bauer <pascal.bauer@rwth-aachen.de>
+ *
+ * SPDX-FileCopyrightText: 2024-25 Pascal Bauer <pascal.bauer@rwth-aachen.de>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <memory>
+
+#include <villas/kernel/devices/linux_driver.hpp>
+#include <villas/kernel/vfio_device.hpp>
+
+namespace villas {
+namespace kernel {
+namespace devices {
+
+class DeviceConnection {
```

Comment:
Could you motivate why we need yet another class to manage devices and VFIO groups, containers?

I already lost track on how many classes we have 😁 

It would be good to understand why they are needed, as it looks like this class in not actually containing the code to bind a device to a VFIO container/group, but instead refers to the VFIO container/group classes?

Reply:
A lot of this is motivated by not breaking the PCIe code. @IgnoreWarnings  can you check if this cannot be integrated into Device?

Reply:
The VFIO API was designed to abstract IOMMU mappings across different types of devices. E.g. platform and PCIe. Hence, the API for attaching devices to VFIO groups and containers is fairly similar.

In my opinion, we should not make it more complicated by implementing this separately for PCIe and Platform devices as this would just mean we lose the generalisation which the Kernel VFIO sub-system is providing us.